### PR TITLE
Add support of DRM_FORMAT_NV12 for console mode rendering

### PR DESCRIPTION
**Pull request by VUNNYSODHI**
Currently in drm/console mode only RGB4 direct rendering
is possible. This patch add NV12 format provided your
platform and kernel can support it.

Issue: #2675

Signed-off: Fu, Wei A <wei.a.fu@intel.com>
Signed-off-by: Sodhi, Vunny <vunny.sodhi@intel.com>

### CODE REVIEW

File: samples/sample_common/src/vaapi_utils_drm.cpp [hunk 1]
```diff
@@ -420,10 +421,45 @@ void* drmRenderer::acquire(mfxMemId mid)
...
+            ret = m_drmlib.drmModeAddFB2(m_fd, vmid->m_image.width, vmid->m_image.height,
+                DRM_FORMAT_NV12, handles, pitches, offsets, &fbhandle, 0);
+        }
+        else
+        {
+            ret = m_drmlib.drmModeAddFB(m_fd,
+            vmid->m_image.width, vmid->m_image.height,
+            24, 32, vmid->m_image.pitches[0],
+            flink_open.handle, &fbhandle);
+        }
```
Comment (dmitryermilov) [inline samples/sample_common/src/vaapi_utils_drm.cpp:462]: @VUNNYSODHI , can we use drmModeAddFB2 for RGB as well?

File: samples/sample_common/src/vaapi_utils_drm.cpp [hunk 2]
```diff
@@ -343,7 +344,7 @@ bool drmRenderer::getPlane()
         if (plane) {
             if (plane->possible_crtcs & (1 << m_crtcIndex)) {
                 for (uint32_t j = 0; j < plane->count_formats; ++j) {
-                    if (plane->formats[j] == DRM_FORMAT_XRGB8888) {
+                    if (plane->formats[j] == DRM_FORMAT_NV12) {
```
Comment (dmitryermilov) [inline samples/sample_common/src/vaapi_utils_drm.cpp:347]: @VUNNYSODHI , can you please explain more about this place? E.g. will RGB continue working after the change?

File: samples/sample_common/src/vaapi_utils_drm.cpp [hunk 3]
```diff
@@ -420,10 +421,45 @@ void* drmRenderer::acquire(mfxMemId mid)
...
+
+            ret = m_drmlib.drmModeAddFB2(m_fd, vmid->m_image.width, vmid->m_image.height,
+                DRM_FORMAT_NV12, handles, pitches, offsets, &fbhandle, 0);
+        }
+        else
+        {
+            ret = m_drmlib.drmModeAddFB(m_fd,
+            vmid->m_image.width, vmid->m_image.height,
+            24, 32, vmid->m_image.pitches[0],
+            flink_open.handle, &fbhandle);
```
Comment (dmitryermilov) [inline samples/sample_common/src/vaapi_utils_drm.cpp:461]: @VUNNYSODHI , can we use drmModeAddFB2 for RGB as well?

**Review by dmitryermilov — COMMENTED** [review]
@VUNNYSODHI , are there kernel versions limitations to use drmModeAddFB2 ?
I'm asking because @dvrogozh has told me that when we added the headless mode rendering in the sample, the kernel didn't support anything beyond RGB. Perhaps we considered  drmModeAddFB only. I don't know. @dvrogozh , appreciate your attempt to recall details.

File: samples/sample_common/src/vaapi_utils_drm.cpp (hunk 2 above)
Comment (weiafu) [reply samples/sample_common/src/vaapi_utils_drm.cpp:347]: We just want to test the NV12 mode, if RGB still needs to be continued working , we can change it as below:

 if ((plane->formats[j] == DRM_FORMAT_XRGB8888) || (plane->formats[j] == DRM_FORMAT_NV12)) {

File: samples/sample_common/src/vaapi_utils_drm.cpp (hunk 3 above)
Comment (weiafu) [reply samples/sample_common/src/vaapi_utils_drm.cpp:461]: No, for RGB, we have to continue using  drmModeAddFB

Comment (dmitryermilov) [reply samples/sample_common/src/vaapi_utils_drm.cpp]: Okay

File: samples/sample_common/src/vaapi_utils_drm.cpp (hunk 2 above)
Comment (dmitryermilov) [reply samples/sample_common/src/vaapi_utils_drm.cpp]: Yes, please make sure RGB continues working.

File: samples/sample_common/src/vaapi_utils_drm.cpp (hunk 3 above)
Comment (dmitryermilov) [reply samples/sample_common/src/vaapi_utils_drm.cpp]: Ohh, I see Vunny enabled both NV12 and RGB via drmModeAddFB2WithModifiers. Cool!

File: samples/sample_common/src/vaapi_utils_drm.cpp (hunk 2 above)
Comment (VUNNYSODHI) [reply samples/sample_common/src/vaapi_utils_drm.cpp]: I have submitted changes, tested  RGB and NV12 both are working fine.

**Review by dmitryermilov — APPROVED** [review]

**Comment by dvrogozh** [comment]
> @dvrogozh , appreciate your attempt to recall details.

At the moment we've implemented this code, there was no AddFB2 or we were not aware of it. And probably this was not a priority to deep dive and find out.

**Review by dvrogozh — APPROVED** [review]

**Pull request merged**